cmd/controller/app/server.go

```diff
@@ -182,19 +182,31 @@ func Run(o *options.ControllerRunOptions, stopCh <-chan struct{}) error {
 		klog.Errorf("unable to setup gc reconciler: %v", err)
 		return err
 	}
+	go func() {
+		for {
+			select {
```

Comment:
why  you need a for and select here? 

can we just block on <-mgr.Elected() and then run anything else?